packages/core/core/src/factories/useForgotPasswordFactory.ts

```diff
@@ -0,0 +1,70 @@
+import { CustomQuery, Context, FactoryParams, UseForgotPasswordErrors, UseForgotPassword } from '../types';
+import { Ref, computed } from '@vue/composition-api';
+import { sharedRef, Logger, configureFactoryParams } from '../utils';
+
+interface IChangePasswordParams {
+  tokenValue: string;
+  newPassword: string;
+}
+
+interface IResetPasswordParams {
+  email: string;
+}
+
+export interface UseForgotPasswordFactoryParams<TOKEN, RESULT> extends FactoryParams {
+  resetPassword: (context: Context, params: IResetPasswordParams & { customQuery?: CustomQuery }) => Promise<TOKEN>;
+  changePassword: (context: Context, params: IChangePasswordParams & { customQuery?: CustomQuery }) => Promise<RESULT>;
+}
+
+export function useForgotPasswordFactory<TOKEN, RESULT>(
+  factoryParams: UseForgotPasswordFactoryParams<TOKEN, RESULT>
+) {
+  return function useForgotPassword(): UseForgotPassword<TOKEN, RESULT> {
+    const result: Ref<RESULT> = sharedRef('', 'useForgotPassword-result');
+    const token: Ref<TOKEN> = sharedRef('', 'useForgotPassword-token');
+    const loading = sharedRef(false, 'useProduct-loading');
+    const _factoryParams = configureFactoryParams(factoryParams);
+    const error: Ref<UseForgotPasswordErrors> = sharedRef({
+      result: null
```

Comment:
`change: null`?

Reply:
As these two methods cannot be used at the same moment (reset in Login Page, change in reset Password page) I decided that there is no need to craete separate errors just for that. Do you think it is necessary to have both methods have separate errors?